base/distributed/managers.jl

```diff
@@ -455,31 +455,36 @@ end
 const client_port = Ref{Cushort}(0)
 
 function socket_reuse_port()
-    s = TCPSocket()
-    client_host = Ref{Cuint}(0)
-    ccall(:jl_tcp_bind, Int32,
-            (Ptr{Void}, UInt16, UInt32, Cuint),
-            s.handle, hton(client_port.x), hton(UInt32(0)), 0) < 0 && throw(SystemError("bind() : "))
+    s = TCPSocket(delay = false)
 
-    # TODO: Support OSX and change the above code to call setsockopt before bind once libuv provides
-    # early access to a socket fd, i.e., before a bind call.
-
-    @static if is_linux()
+    @static if is_linux() || is_apple()
         try
+            is_linux() && bind_client_port(s)
             rc = ccall(:jl_tcp_reuseport, Int32, (Ptr{Void},), s.handle)
             if rc > 0  # SO_REUSEPORT is unsupported, just return the ephemerally bound socket
                 return s
             elseif rc < 0
                 throw(SystemError("setsockopt() SO_REUSEPORT : "))
             end
-            getsockname(s)
+            is_apple() && bind_client_port(s)
         catch e
             # This is an issue only on systems with lots of client connections, hence delay the warning
             nworkers() > 128 && warn_once("Error trying to reuse client port number, falling back to plain socket : ", e)
```

Comment:
why is this done as a try/catch rather than printing this message where we through it 6 lines above and then returning from there